logstash-core/lib/logstash/webserver.rb

```diff
@@ -105,6 +106,14 @@ def self.required_setting(settings, setting_name, trigger)
       settings.get(setting_name) || fail(ArgumentError, "Setting `#{setting_name}` is required when `#{trigger}` is set to `#{settings.get(trigger)}`. Please provide it in your `logstash.yml`")
     end
 
+    def self.required_setting_with_changing_default(settings, name, trigger, future_value)
+      effective_value = required_setting(settings, name, trigger)
+      if !settings.set?(name)
+        deprecation_logger.deprecated("The default value of `#{name}` will change to `#{future_value}` in a future release of Logstash. Set it to `#{future_value}` to enforce future requirement now.")
```

Comment:
A deprecation log should not only inform a user of an upcoming breaking change, it should empower them to address it by providing a clear path to either consuming the breaking change early or avoiding the breaking change by explicitly locking in the current behavior.
```suggestion
        deprecation_logger.deprecated("The default value of `#{name}` will change to `#{future_value}` in a future release of Logstash. Set it to `#{future_value}` to observe the future behavior early, or set it to `#{effective_value}` to lock in the current behavior.")
```